specs/withdrawals.md

```diff
@@ -0,0 +1,176 @@
+
+1. The `WithdrawalsRootInclusionProof` proof contains the data necessary to prove that the provided `WithdrawalsRoot` is
+  included in the [SSZ merkleization](https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md#merkleization)
+  of the `L2Output` as defined in the
+  [L2 output commitment construction](./proposals.md#l2-output-commitment-construction).
+
+1. A `WithdrawalMessageInclusionProof` is an MPT proof encoded in a struct as follows:
+
+   ```js
+   struct WithdrawalMessageInclusionProof {
+       WithdrawalMessage message;
+       bytes32 l2withdrawalsRoot;
+       bytes memory _key;      // storage key of the withdrawal message commitment
+       bytes memory _value;    // Always bytes32(1) (boolean validity status of the message)
+       bytes memory _proof;    // MPT inclusion proof for the key/value
+   }
+   ```
+1. This document also refers to the `L2Output` type as defined in the
+  [L2 output commitment construction](./proposals.md#l2-output-commitment-construction).
+
+## Withdrawal verification
+
+In order to verify a `WithdrawalMessage`, the following inputs must be provided:
+
+| Type                              | Name               |
+| --------------------------------- | ------------------ |
+| `WithdrawalMessage`               | `message`          |
+| `L2OutputTimestamp`               | `timestamp`        |
+| `WithdrawalsRootInclusionProof`   | `storageRootProof` |
+| `WithdrawalMessageInclusionProof` | `messageProof`     |
+
+These inputs must satisfy the following conditions:
+
+1. The `timestamp` is at least `FINALIZATION_WINDOW` seconds old.
+1. `OutputOracle.l2Outputs(timestamp)` returns a non-zero value `l2Output`.
+1. The `storageRootProof` is valid for the `l2Output` returned in step 2, according to the SSZ encoding described in the
+   [L2 output commitment construction](./proposals.md#l2-output-commitment-construction).
+1. The `messageProof` is valid for the provided `message` and `messageProof`
+
+## Withdrawal Flow
+
+**On L2:**
+
+1. An L2 account sends a withdrawal message (and possibly also ETH) to the `Withdrawoooor` predeploy contract.
+   This is a very simple contract that stores a mapping from the hash of the `WithdrawalMessage` as defined above to a
+   boolean value. (`mapping (bytes32 => bool) withdrawalMessages`)
+2. If ETH is being withdrawn, it can eventually be burned by deploying a contract which immediately `SELFDESTRUCT`s.
+   This has the benefit of using an existing EVM mechanism for removing the ETH from the world state, without having to
+   add a diff to L2 Geth.
+
+**On L1:**
+
+1. A [relayer][g-relayer] submits the required inputs to the `DepositFeed` contract. The relayer may or may not be the same entity which
+   initiated the withdrawal on L2.
+2. The `DepositFeed` contract retrieves the output root from the `OutputOracle`'s `l2Outputs()` function, and performs
+   the remainder of the verification process internally.
```

Comment:
Would put a forward reference here, otherwise you're wondering what the rest of the process is and if it's already been explained. (But it's not yet since this is still a draft.)